src/protocols/PresentationTime.cpp

```diff
@@ -58,9 +58,15 @@ void CPresentationFeedback::sendQueued(SP<CQueuedPresentationData> data, timespe
     if (reportedFlags & Aquamarine::IOutput::AQ_OUTPUT_PRESENT_HW_COMPLETION)
         flags |= WP_PRESENTATION_FEEDBACK_KIND_HW_COMPLETION;
 
+    auto tv_sec = (uint32_t)when->tv_sec;
+
+#ifdef __LP64__ //64-bit architecture check
+    tv_sec = (uint32_t)(when->tv_sec >> 32);
+#endif
+
     if (data->wasPresented)
-        resource->sendPresented((uint32_t)(when->tv_sec >> 32), (uint32_t)(when->tv_sec & 0xFFFFFFFF), (uint32_t)(when->tv_nsec), untilRefreshNs, (uint32_t)(seq >> 32),
-                                (uint32_t)(seq & 0xFFFFFFFF), (wpPresentationFeedbackKind)flags);
+        resource->sendPresented(tv_sec, (uint32_t)(when->tv_sec & 0xFFFFFFFF), (uint32_t)(when->tv_nsec), untilRefreshNs, (uint32_t)(seq >> 32), (uint32_t)(seq & 0xFFFFFFFF),
```

Comment:
this looks wrong indeed

Reply:
pretty much this that had me worried https://godbolt.org/z/rKfovscxz but no idea

Reply:
i mean isnt this trying to obtain the upper 32bits of the 64bit tv_sec type, only on 32bit it isnt a 64bit type. meaning we cant do it there and just pass it the tv_sec directly?

Reply:
I'd just leave it as-is

Reply:
its gonna break on 32bit tho , i mean i could technically even do something like 

```
if (sizeof(__time_t) > 4)
    tv_sec = (uint32_t)(when->tv_sec >> 32);
```

i think

Reply:
or should it even be 0 if there is no upper bits hm

Reply:
since the next argument is taking the lower bits and as on 32bit just passes it along as is

Reply:
something like this maybe. idk, otherwise its gonna bork. as the gcc warning hinted when compiling its UB and the value turned out to something like -51237 or similiar on the 32bit build i did.

```
    __time_t tv_sec = 0;

    if (sizeof(__time_t) > 4)
        tv_sec = (uint32_t)(when->tv_sec >> 32);

    if (data->wasPresented)
        resource->sendPresented(tv_sec, (uint32_t)(when->tv_sec & 0xFFFFFFFF), (uint32_t)(when->tv_nsec), untilRefreshNs, (uint32_t)(seq >> 32), (uint32_t)(seq & 0xFFFFFFFF),
                                (wpPresentationFeedbackKind)flags);
```

Reply:
uh huh, I guess ok

Reply:
ok changed, but could really use a test on some 32bit system if this works as intended heh